[SlothyCat] iP

SlothyCat opened the pull request:
## Nimbus
> "If you fail to plan, you plan to fail" - Benjamin Franklin ([source](https://www.goodreads.com/quotes/460142-if-you-fail-to-plan-you-are-planning-to-fail))

Nimbus keeps track of what you need to do

- text-based
- easy to learn
- Super fast to use

All you need to do is,

1. download it from [here](https://github.com/SlothyCat/ip)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😀 

And it is **FREE**!

Features:
- [x] Managing Tasks
- [x] Managing deadlines 
- [ ] Reminders (coming soon)

---
If you are a Java programmar, you can use it to practice Java too. Here's the `main` method

```java
public static void main(String[] args)
        throws IOException, WrongDateTimeFormatException {
    new Nimbus("nimbus.txt").run();
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SlothyCat committed:
Completed Level 0

SlothyCat committed:
Completed Level 1

SlothyCat committed:
Completed Level 2

SlothyCat committed:
Completed Level 3

SlothyCat committed:
Completed Level 4

SlothyCat committed:
Completed A-TextUiTesting

SlothyCat committed:
Completed Level 5

SlothyCat committed:
Completed Level 6

SlothyCat committed:
Complete Level 7

SlothyCat committed:
complete Level 7

SlothyCat committed:
Merge branch 'branch-Level-7'

SlothyCat committed:
add Datetime

SlothyCat committed:
fix event toFileFormat

SlothyCat committed:
add DateTime for Deadline Task

SlothyCat committed:
complete Level 8

SlothyCat committed:
Merge branch 'branch-Level-8'

SlothyCat committed:
amend conflicted files

SlothyCat committed:
Complete A-MoreOOP

SlothyCat committed:
Complete Stretch Goal for MoreOOP

SlothyCat committed:
Complete A-Packages

SlothyCat committed:
Tidy up Nimbus.java

SlothyCat committed:
Merge branch 'add-gradle-support'

SlothyCat committed:
Complete A-Gradle

SlothyCat committed:
Complete A-JUnit

SlothyCat committed:
Add Jar Config

SlothyCat committed:
Update README.md

SlothyCat committed:
Complete JAR file

SlothyCat committed:
Complete A-JavaDoc

SlothyCat committed:
Complete A-CodingStandard

SlothyCat committed:
Complete Level-9

SlothyCat committed:
Merge branch 'branch-A-JavaDoc'

SlothyCat committed:
Merge branch 'branch-A-CodingStandard'

SlothyCat committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/nimbus/task/Task.java

SlothyCat committed:
Fix conflicted files

SeanFoongjt commented on src/main/java/nimbus/command/DeleteCommand.java:
```diff
@@ -0,0 +1,56 @@
+            String temp = tasks.get(index).toString();
+            Task task = tasks.get(index);
+            taskList.delete(task);
+            System.out.println("Nimbus.Nimbus has removed the task! \n" +
```
Should this operator (+) be on the next line?

SlothyCat replied:
Yup it should have been! Thank youu 😄

SeanFoongjt commented on src/main/java/nimbus/command/FindCommand.java:
```diff
@@ -0,0 +1,56 @@
+    @Override
+    public void execute() {
+        if (userInput.length() < 6) {
+            System.out.println("Please enter what you want to find" +
```
Should this operator (+) be on the next line? I think the problem is around for some other places too.

SeanFoongjt commented on src/main/java/nimbus/ui/Ui.java:
```diff
@@ -0,0 +1,78 @@
+            "|  .     |/    |  |(_/  |  |'.'|  |  | .-. `.   |  |_|( OO ) '..`''.) \n" +
+            "|  |\\    |    ,|  |_.'  |  |   |  |  | |  \\  |  |  | | `-' /.-._)   \\ \n" +
+            "|  | \\   |   (_|  |     |  |   |  |  | '--'  / ('  '-'(_.-' \\       / \n" +
+            "`--'  `--'     `--'     `--'   `--'  `------'    `-----'     `-----'  \n";
```
I like the logo!

SlothyCat replied:
Thank you! You can get the logo from the website I commented too

SeanFoongjt commented on src/test/java/nimbus/command/AddCommandTest.java:
```diff
@@ -0,0 +1,47 @@
+import org.junit.jupiter.api.Test;
+
+
+import static org.junit.jupiter.api.Assertions.*;
```
Perhaps this could instead be explicit imports of Assertion classes.

SeanFoongjt commented on src/main/java/nimbus/command/AddCommand.java:
```diff
@@ -0,0 +1,111 @@
+     * @throws MissingStartEndTimeException if there is either missing start or end time for events
+     * @throws WrongDateTimeFormatException if dateTime format is wrong in userInput
+     */
+
```
Should there be a blank line between the documentation block and the method? This may be a problem elsewhere in the code also.

SeanFoongjt commented on src/main/java/nimbus/ui/Storage.java:
```diff
@@ -0,0 +1,163 @@
+
+public class Storage {
+    private static String filepath;
+    private final Logger logger = Logger.getLogger(Nimbus.class.getName());
```
Perhaps this could have a name like dynamicLogger to match the staticLogger.

SeanFoongjt reviewed (commented):
LGTM! I think the main issue I see is the empty lines between the documentation blocks and the methods/classes, but otherwise everything looks pretty neat!

hyxnnii commented on src/test/java/nimbus/command/DeleteCommandTest.java:
```diff
@@ -0,0 +1,52 @@
+
+public class DeleteCommandTest {
+    @Test
+    public void testDeleteCommandOneTask() {
```
perhaps can use the other convention instead of camel case for test case names? (unitBeingTested_descriptionOfTestInputs_expectedOutcome)

hyxnnii reviewed (commented):
overall I think you did a great job :-)

SlothyCat committed:
Complete A-CheckStyle

SlothyCat committed:
Complete Level-10 GUI

SlothyCat committed:
Merge branch 'branch-Level-10'

SlothyCat committed:
Recreate JAR File

SlothyCat committed:
Complete A-Assertions

Certain classes lack assertions to ensure code flow.

Assertions have been implemented in command classes
to make sure that the assumptions made are always true
in the command classes.

Assertions have been implemented with messages that
describe what were the assumptions made

SlothyCat committed:
Complete A-CodeQuality

Some methods are too long and can be better abstracted
using the SLAP.

Keeping each method as simple as possible through
refactoring, mainly done on execute in addCommand
and execute in checkCommand

This makes the methods easier to understand and show
the different steps that made them work.

SlothyCat committed:
Merge pull request #2 from SlothyCat/branch-A-Assertions

Complete A-Assertions

SlothyCat committed:
Merge branch 'master' into branch-A-CodeQuality

SlothyCat committed:
Merge pull request #3 from SlothyCat/branch-A-CodeQuality

Complete A-CodeQuality

SlothyCat committed:
Complete B-ViewSchedules

User will need a way to find tasks that happen on a
particular day.

A Check Command can help users to check for any
tasks happening on the specific day.

It is similar to how other commands work, just with
a check keyword now.

SlothyCat committed:
Change Exception for CheckCommand

Previous exception was not accurate, giving
the wrong date time format message when
exception is called.

New exception has been created to better suit
the CheckCommand.

SlothyCat committed:
Complete A-BetterGui

SlothyCat committed:
Complete A-Personality

SlothyCat committed:
Refine JavaDocs and remove unnecessary assets

SlothyCat committed:
Update README.md

SlothyCat committed:
Update README.md

SlothyCat committed:
Update README.md

SlothyCat committed:
Update tests to follow code standards

SlothyCat committed:
Update README.md

SlothyCat committed:
Update README.md

SlothyCat committed:
Complete Final Touch ups
